services/knack_street_seg_updater.py

```diff
@@ -144,10 +144,14 @@ def main(args):
         features = query_atx_street(
             street_segment[config["primary_key"]], token
         )
-        time.sleep(1) # wait one second between requests to prevent being rate limited
         if features.get("error"):
             # AGOL returns code 200 even for error queries.
-            raise Exception(str(features))
+            # Let's try the query again
```

Comment:
this works just fine, but might be worth setting up a loop in case we want to tweak retries? e.g. `while not features and attemps < MAX_ATTEMPTS ...`